shared/common-adapters/form-input.native.js

```diff
@@ -0,0 +1,165 @@
+// @flow
+import * as React from 'react'
+import {Box, Text} from '.'
+import type {Props} from './form-input'
+import {NativeTextInput} from './native-wrappers.native'
+import {globalColors, globalMargins, globalStyles} from '../styles'
+import {getStyle} from './text'
+
+type State = {
+  value: string,
+  focused: boolean,
+}
+
+export const backgroundModeToColor = {
+  Normal: globalColors.white,
+  Terminal: globalColors.darkBlue3,
+  Announcements: globalColors.blue,
+  Success: globalColors.green,
+  Information: globalColors.yellow,
+  HighRisk: globalColors.red,
+  Documentation: globalColors.darkBlue,
+}
+
+class FormInput extends React.Component<Props, State> {
+  state: State
+  _input: any
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      value: this.props.value || '',
```

Comment:
i think you can use defaultProps for this so value is always a string
then you can just remove this constructor and make state as an initialized member property